boot/bootstate20_bloader_kernel_state.go

```diff
@@ -185,6 +185,11 @@ func (bks *extractedRunKernelImageBootloaderKernelState) setNextKernelNoTry(sn s
 		}
 	}
 
+	// Make sure that no try-kernel.efi link is left around. We do
```

Comment:
Is it possible to unit test this and the above change somehow?

Reply:
I've added a unit test now

Reply:
the new unit test covers also the change in bootstate20.go, right?

Reply:
Yes, as we check that there is only one kernel in current_kernels (before the fix we were storing two kernels).